### PR TITLE
Refactored assets CE + tests

### PR DESCRIPTION
**Pull request by PierreOssun**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,8 +1,6 @@
 [workspace]
 members = [
-    "examples/dapps-staking",
-#    "examples/rmrk",
-#    "examples/rooster-contract",
+    #     "examples/dapps-staking",
```
Comment (ashutoshvarma) [inline Cargo.toml]:
```suggestion
    # "examples/dapps-staking",
```

or remove it?

File: contracts/psp22_pallet_wrapper/lib.rs [hunk 2]
```diff
@@ -74,11 +74,12 @@ pub mod psp22_pallet_wrapper {
             self.asset_id
         }
 
+        /// Caller should approve contract address as spender beforehand
         #[ink(message)]
         pub fn deposit(&mut self, amount: Balance) -> Result<(), PSP22Error> {
             let caller = self.env().caller();
             let contract = self.env().account_id();
-            AssetsExtension::transfer(Origin::Caller, self.asset_id, contract, amount)
+            AssetsExtension::transfer_approved(Origin::Address, self.asset_id, caller, contract,  amount)
```
Comment (ashutoshvarma) [inline contracts/psp22_pallet_wrapper/lib.rs]:
```suggestion
            AssetsExtension::transfer_approved(Origin::Address, self.asset_id, caller, contract, amount)
```
`cargo fmt --all`

File: contracts/psp22_pallet_wrapper/Cargo.toml [hunk 3]
```diff
@@ -5,16 +5,15 @@ authors = ["Stake Technologies <devops@stake.co.jp>"]
...
 [dependencies]
-ink = { version = "~4.0.0-rc", default-features = false}
-ink_metadata = { version = "~4.0.0-rc", features = ["derive"], optional = true }
+ink = { version = "~4.0.0", default-features = false}
+ink_metadata = { version = "~4.0.0", features = ["derive"], optional = true }
 
 scale = { package = "parity-scale-codec", version = "3", default-features = false, features = ["derive"] }
 scale-info = { version = "2", default-features = false, features = ["derive"], optional = true }
 
 openbrush = { git = "https://github.com/727-Ventures/openbrush-contracts", version = "3.0.0-beta.1", default-features = false, features = ["psp22"] }
```
Comment (ashutoshvarma) [inline contracts/psp22_pallet_wrapper/Cargo.toml]:
```suggestion
openbrush = { git = "https://github.com/727-Ventures/openbrush-contracts", version = "3.0.0", default-features = false, features = ["psp22"] }
```

File: contracts/psp22_pallet_wrapper/Cargo.toml [hunk 4]
```diff
@@ -5,16 +5,15 @@ authors = ["Stake Technologies <devops@stake.co.jp>"]
 edition = "2021"
 
 [dependencies]
-ink = { version = "~4.0.0-rc", default-features = false}
-ink_metadata = { version = "~4.0.0-rc", features = ["derive"], optional = true }
+ink = { version = "~4.0.0", default-features = false}
+ink_metadata = { version = "~4.0.0", features = ["derive"], optional = true }
```
Comment (ashutoshvarma) [inline contracts/psp22_pallet_wrapper/Cargo.toml]:
```suggestion
ink = { version = "~4.0.0", default-features = false }
```

No need to add `ink_metadata` directly as `ink/std` includes it.
https://github.com/paritytech/ink/blob/bb9cc8cb624c98e4e273b99142c6300a5e399dd3/crates/ink/Cargo.toml#L40

File: crates/assets/Cargo.toml [hunk 5]
```diff
@@ -5,12 +5,11 @@ authors = ["Stake Technologies <devops@stake.co.jp>"]
 edition = "2021"
 
 [dependencies]
-ink = { version = "4.0.0-rc", default-features = false }
-ink_metadata = { version = "4.0.0-rc", features = ["derive"], optional = true }
+ink = { version = "~4.0.0", default-features = false }
+ink_metadata = { version = "~4.0.0", features = ["derive"], optional = true }
```
Comment (ashutoshvarma) [inline crates/assets/Cargo.toml]:
```suggestion
ink = { version = "~4.0.0", default-features = false }
```
No need to add ink_metadata directly as ink/std includes it.
https://github.com/paritytech/ink/blob/bb9cc8cb624c98e4e273b99142c6300a5e399dd3/crates/ink/Cargo.toml#L40

File: crates/assets/Cargo.toml [hunk 6]
```diff
@@ -24,5 +23,4 @@ std = [
     "ink_metadata/std",
```
Comment (ashutoshvarma) [inline crates/assets/Cargo.toml]:
```suggestion
```

File: crates/dapps-staking/Cargo.toml [hunk 7]
```diff
@@ -5,8 +5,8 @@ authors = ["Stake Technologies <devops@stake.co.jp>"]
 edition = "2021"
 
 [dependencies]
-ink = { version = "4.0.0-rc", default-features = false }
-ink_metadata = { version = "4.0.0-rc", features = ["derive"], optional = true }
+ink = { version = "~4.0.0", default-features = false }
+ink_metadata = { version = "~4.0.0", features = ["derive"], optional = true }
```
Comment (ashutoshvarma) [inline crates/dapps-staking/Cargo.toml]:
```suggestion
ink = { version = "~4.0.0", default-features = false }
```

File: examples/assets/Cargo.toml [hunk 8]
```diff
@@ -5,12 +5,11 @@ authors = ["Stake Technologies <devops@stake.co.jp>"]
 edition = "2021"
 
 [dependencies]
-ink = { version = "4.0.0-rc", default-features = false }
-ink_metadata = { version = "4.0.0-rc", features = ["derive"], optional = true }
+ink = { version = "~4.0.0", default-features = false }
+ink_metadata = { version = "~4.0.0", features = ["derive"], optional = true }
```
Comment (ashutoshvarma) [inline examples/assets/Cargo.toml]:
```suggestion
ink = { version = "~4.0.0", default-features = false }
```

File: examples/dapps-staking/Cargo.toml [hunk 9]
```diff
@@ -5,8 +5,8 @@ authors = ["Stake Technologies <devops@stake.co.jp>"]
 edition = "2021"
 
 [dependencies]
-ink = { version = "4.0.0-rc", default-features = false }
-ink_metadata = { version = "4.0.0-rc", features = ["derive"], optional = true }
+ink = { version = "~4.0.0", default-features = false }
+ink_metadata = { version = "~4.0.0", features = ["derive"], optional = true }
```
Comment (ashutoshvarma) [inline examples/dapps-staking/Cargo.toml]:
```suggestion
ink = { version = "~4.0.0", default-features = false }
```

File: tests/dapps-staking.spec.ts
Comment (ashutoshvarma) [inline tests/dapps-staking.spec.ts:1]: Might as well remove it?

Comment (Maar-io) [reply tests/dapps-staking.spec.ts:1]: Don't delete, let me open an issue and update tests

**Review by ashutoshvarma — APPROVED** [review]

**Pull request merged**